docs/4.4/enterprise/workflow/index.md

```diff
@@ -129,6 +129,87 @@ deny:
     roles: ['admin']
 ```
 
+## Adding a reason to approval requests
+
+Teleport 4.4.4 introduced the ability for users to request additional roles. `tctl`
+or the Access Workflow API makes it easy to dynamically approve or deny these requests.
+
+In the Supporting users that start in an essentially unprivileged state and must
+always go through the dynamic access API in order to gain meaningful privilege.
+
+Leveraging the claims (traits) provided by external identity providers both when
+determining which roles a user is allowed to request, and if a specific request
+should be approved/denied.
+
+### Example Setup
+
+**Unprivileged User**<br>
+In this example we've a employee, who isn't able to access any systems. When they
+login they'll always need to provide a reason to access.
+
+```yaml
+kind: role
+metadata:
+  name: employee
+spec:
+  allow:
+    request:
+      # the `roles` list can now be a mixture of literals and matchers
+      roles: ['common', 'dev-*']
+      # the `claims_to_roles` mapping works the same as it does in
+      # the oidc connector, with the added benefit that the mapped to roles
+      # can also be matchers.  the below mapping says that users with
+      # the claims `groups: admins` can request any role in the system.
+      claims_to_roles:
+        - claim: groups
+          value: admins
+          roles: ['*']
+      # teleport can attach annotations to pending access requests. these
+      # annotations may be literals, or be variable interpolation expressions,
+      # effectively creating a means for propagating selected claims from an
+      # external identity provider to the plugin system.
+      annotations:
+        foo: ['bar']
+        groups: ['{% raw %}{{external.groups}}{% endraw %}']
+  options:
+    # the `request_access` field can be set to 'always' or 'reason' to tell
+    # tsh of the web UI to always create an access request on login.  If it is
+    # set to 'reason', the user will be required to indicate *why* they are
+    # generating the access request.
+    request_access: reason
+    # the `request_prompt` field can be used to tell the user what should
+    # be supplied in the request reason field.
+    request_prompt: Please provide your ticket ID
+version: v3
+```
+
+**Unprivileged User Login**<br>
+
+```bash
+# Login: This will prompt the user to provide a reason in the UI.
+tsh login
+# Login: The user can provide a reason using tsh.
+tsh login --request-reason="..."
+```
+
+!!! Note
+
+    Notice that the above role does not specify any logins. If a users's roles specify no logins, teleport will now generate the user's initial SSH certificates with an invalid dummy login of the form `-teleport-nologin-<uuid>` (e.g. `-teleport-nologin-1e02dbfd-8f6e-47a0-a66c-93747b010f88`).
```

Comment:
Incidentally, this login will be invalid because it's longer than the constant which defines max login length and as such it will generate some log spam. rjones' AAP PR suffered from the same thing. We should either raise the constant to account for this or stop using UUIDs as invalid logins.

Reply:
@fspmarshall Do you have any thoughts on this?

Reply:
Oops, I hadn't noticed.  I'd like to include some randomness, but it doesn't need to be a full UUID.  Something like `-teleport-nologin-1e02dbfd` would work just fine.  For now, @benarent maybe just change the docs to just specify `-teleport-nologin-<random-suffix>` without giving an example?

Reply:
Fixed in last commit